### PR TITLE
feat(Entity): PFR-59 - Endpoint Masterclass Entity

### PR DESCRIPTION
**Pull request by tgamiette**
| Q  | A |
| --- | --- |
| Type | feature |
| Related Tickets | [PFR-55](https://project-fil-rouge.atlassian.net/browse/PFR-55) |

## What does this PR do?
endpoint MAsterclass User

- [ ] Unit tests
- [ ] Behat tests

[PFR-55]: https://project-fil-rouge.atlassian.net/browse/PFR-55?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/back/src/Entity/MasterclassUser.php [hunk 1]
```diff
@@ -3,49 +3,80 @@
...
+                    fromClass: MasterclassUser::class
+                ),
+            ],
+            requirements: ['masterclassId' => '\d+'],
+            security: "is_granted('MASTERCLASS_USER_CREATE', object)",
+            processor: MasterclassUserStateProcessor::class
+        ),
+        //                processor: MasterclassUserStateProcessor::class),
+        //        new Delete(security: "is_granted('MASTERCLASS_USER_DELETE', object)"),
+        //        new Patch(securityPostDenormalize: "is_granted('MASTERCLASS_USER_DELETE', [object, previous_object])"),
```
Comment (Sebastien-jo) [inline app/back/src/Entity/MasterclassUser.php:46]: on peut supprimer ça ?

File: app/back/src/State/MasterclassUserStateProcessor.php [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            $event = new MasterclassUserEvent($masterclass, $user);
+
+            if ($operation instanceof DeleteOperationInterface) {
+                return $this->dispatcher->dispatch($event, MasterclassUserEvent::DELETE);
+            }
+
+            return $this->dispatcher->dispatch($event, MasterclassUserEvent::CREATE);
+        } catch (Exception $e) {
+            $this->logger->error($e->getMessage());
+            throw new Exception($e->getMessage(), $e->getCode(), $e);
```
Comment (Sebastien-jo) [inline app/back/src/State/MasterclassUserStateProcessor.php:59]: pourquoi log et ensuite throw ?

**Comment by Sebastien-jo** [comment]
une erreur aussi :
`  A binding is configured for an argument named "$persistProcessor" for service "App\State\MasterclassUserStateProcessor" in file "/var/www/symfony/config/services.yaml", but no corres  
  ponding argument has been found. It may be unused and should be removed, or it may have a typo. `

File: app/back/src/State/MasterclassUserStateProcessor.php (hunk 2 above)
Comment (tgamiette) [reply app/back/src/State/MasterclassUserStateProcessor.php:59]: log pour logger dans les log 
throw pour  gnérer une erreur avec l'api

**Pull request merged**